content/luau/concepts/arrays/arrays.md

```diff
@@ -0,0 +1,118 @@
+---
+Title: 'Arrays'
+Description: 'Arrays are used to store ordered collections of data.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Tables'
+  - 'Arrays'
+  - 'Iterators'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+**Arrays** are a type of container for storing data in ordered collections. Luau, a lightweight and efficient scripting language, implements arrays using a single structure known as a table.
+
+Arrays are versatile data structures capable of storing various data types, and values can be retrieved or modified using the conventional indexing syntax `[]`. To clear an array, the elements are set to `nil`.
+
+## Syntax
+
+Luau allows for the creation of arrays using curly braces `{}`. Unlike some programming languages that start array indexing from `0`, in Luau, the convention is to begin with index `1`. By default, the first element in an array is designated as index `1`. However, Luau provides the flexibility to create arrays with custom indices as needed.
+
+```lua
+a = {5};  -- An array with single elemnent
+
+print(a[1]) -- This will return 5
+print(a[2]) -- This will return nil because the index is out of range
+```
+
+The array created above contains one element only with the value `5`, it can be accessed by passing the index value `1`. However, if a value is passed out of the range of the array , the code will return `nil`. This is because `nil` refers to missing data.
+
+## Accessing Elements
+
+To access an array element, the desired value is called by its _index_. For example, to reference a value from the following array:
+
+```lua
+a = {
+    [1] = “a”,
+    [2] = “b”,
+    [3] = “c”
+}
+```
+
+In this array, elements are indexed from 1 to 3. To print `b`, you need to reference it by its assigned index, which is 2.
+
+This indexing method works for all elements in an array. This also works with dictionaries, but instead the assigned _key_ must be used to retrieve a value. A _key_ is essentially an _index_ with the only difference being that a _key_ may be assigned as any type, while an _index_ must be a number.
+
+For example:
+
+```lua
+dictionary = {
+    [“A”] = “Apple”,
+    [“B”] = “Banana”,
+    [“C”] = “Cat”
+}
+
+print(dictionary[“C”]) -- “Cat”
+```
```

Comment:
````suggestion
```lua
dictionary = {
    ["A"] = "Apple",
    ["B"] = "Banana",
    ["C"] = "Cat"
}

print(dictionary["C"]) -- "Cat"
```
````
Regular straight quotes should be used in code.